tensorflow/compiler/mlir/python/mlir.cc

```diff
@@ -133,6 +148,87 @@ std::string ImportFunction(const std::string &functiondef_proto,
                                  status);
 }
 
+std::string ImportGraphDef(const std::string& proto,
+                           const std::string& pass_pipeline,
+                           bool show_debug_info, TF_Status* status) {
+  GraphDebugInfo debug_info;
+  GraphImportConfig specs;
+  return ImportGraphDefImpl(proto, pass_pipeline, show_debug_info, debug_info,
+                            specs, status);
+}
+
+std::string ImportGraphDef(const std::string& proto,
+                           const std::string& pass_pipeline,
+                           bool show_debug_info, absl::string_view input_names,
+                           absl::string_view input_data_types,
+                           absl::string_view input_data_shapes,
+                           absl::string_view output_names, TF_Status* status) {
+  std::vector<string> node_names = absl::StrSplit(input_names, ',');
+  std::vector<string> node_dtypes = absl::StrSplit(input_data_types, ',');
+  std::vector<string> node_shapes_str = absl::StrSplit(input_data_shapes, ':');
+  std::vector<std::vector<int>> node_shapes;
+
+  std::vector<int> dims;
+  for (unsigned i = 0, e = node_shapes_str.size(); i < e; i++) {
+    dims.clear();
+    for (const auto& dim_str : absl::StrSplit(node_shapes_str[i], ',')) {
+      // Treats empty input shape as scalar.
+      if (dim_str.empty()) continue;
+      int size;
+      absl::SimpleAtoi(dim_str, &size);
+      dims.push_back(size);
+    }
+    node_shapes.push_back(dims);
+  }
+  std::vector<string> output_nodes = absl::StrSplit(output_names, ',');
```

Comment:
Could all this logic be implemented in python instead? Have the API take a python list instead of doing string manipulation in C++?

Reply:
For functions like `SavedModelToMlir` the string manipulation is done in cpp.

Reply:
I implemented that to be consistent. I think python implementation would be more compact.

Reply:
If you agree that the python implementation makes sense here, let's do it.

We should likely migrate the `SavedModelToMlir` as well.

Reply:
But it's probably not great to have inconsistency b/w the paths - i.e., one API using the list and the other comma separated ones? I think both should be migrated in one shot if that's the better interface -- that's better from a git history/change standpoint as well?

Reply:
If you prefer to avoid introducing inconsistency, then please migrate the API first, then rebase this patch on top of it :)

Reply:
@joker-eph @bondhugula During the implementation, I realized we can have this minor string manipulation over here instead of doing it at different places. Also, we need to populate the GraphImportConfig over here. I think if the whole logic is present over one place it would be nice to debug/understand.